### ISSUE TITLE
Should GenerateCommunicationThrow be removed?

### ISSUE THREAD

**Issue by wo80** [avatar29A/MusicBrainz#38]
I think, swallowing exceptions is never a good idea, so:

Should `GenerateCommunicationThrow` be removed from `Configuration` class?

**Comment by wo80** [comment]
In my current working branch, if the request fails a [WebserviceException](https://github.com/wo80/MusicBrainz/blob/examples/Hqub.MusicBrainz/Hqub.MusicBrainz.API/WebserviceException.cs) will be thrown and you can check the HTTP status code (in case of rate limit: 503).

**Comment by ColorTwist** [comment]
I get you,
however, when this error happen, it breaks the program.
For example, when calling: `var artist= Artist.Get(ArtistMbid);`
If it reaches the musicbrainz limit the exception WebRequestHelper rises from MusicBrainz dll, but i won't receive notification about it in program it self.

**Comment by ColorTwist** [comment]
Oh u mean, to  catch (WebServiceException ex) the queries.

**Comment by wo80** [comment]
Yes, when using async-await this should do it. If you are doing other fancy stuff with tasks (like the above `Artist.Get(ArtistMbid)`, which uses task.Result), you will have to watch for [AggregateException](https://msdn.microsoft.com/en-us//library/dd321468(v=vs.110).aspx).

**Comment by ColorTwist** [comment]
I see, that's great, i missed the "AggregateException".
I never used "Tasks" before, i did not know they generates different error exceptions type.

**Comment by ColorTwist** [comment]
yes, i do use it, and going to remove it and change it to Task based  (GetAsync).
Maybe u should remove it next update :)
I saw you added examples which is great.

**Comment by avatar29A** [comment]
In general, I agree. But , to my mind this update can break backward compatibility. In current time project don't have any clear version policy.

**Comment by wo80** [comment]
Well, it definitly breaks backward compatibility. But since the next nuget release will probably be 2.0, breaking changes should be ok.

**Comment by ColorTwist** [comment]
Note that with ReleaseGroup.BrowseAsync it returns HttpRequestException with rate limit.
A bit confusing that each function in API can return different exceptions.

**Comment by wo80** [comment]
Are you using the latest code from my branch? I do get `WebServiceException` 503 with
```
Your requests are exceeding the allowable rate limit. Please see
http://wiki.musicbrainz.org/XMLWebService for more information.
```
Test code:
```c#
private static void Test()
{
    var mbid = "b071f9fa-14b0-4217-8e97-eb41da73f598"; // Rolling Stones

    int offset = 0, limit = 10;

    int total = 51;

    var tasks = new Task[total];

    for (int i = 0; i < total; i++)
    {
        tasks[i] = ReleaseGroup.BrowseAsync("artist", mbid, limit, offset);

        offset += limit;
    }

    Task.WaitAll(tasks);
}
```

**Comment by wo80** [comment]
```bash
git clone https://github.com/wo80/MusicBrainz.git
cd MusicBrainz
git checkout --track origin/examples
```

**Comment by ColorTwist** [comment]
git merge --abort
fixed it.

**Comment by ColorTwist** [comment]
Maybe merge examples with master :) any new user will need it.

**Comment by wo80** [comment]
That doesn't work so well with @avatar29A 's habit of squashing pull requests ;-)

**Comment by wo80** [comment]
@avatar29A 
Any thoughts on this one? I could add the change to the current PR.

**Comment by wo80** [comment]
Implemented in #51.
